kube_hunter/modules/discovery/hosts.py

```diff
@@ -168,14 +155,13 @@ def execute(self):
             except ValueError as e:
                 logger.exception(f"Unable to parse CIDR \"{config.cidr}\"")
                 return
-            cloud = HostDiscoveryHelpers.get_cloud(ip)
             for ip in HostDiscoveryHelpers.generate_subnet(ip, sn=sn):
-                self.publish_event(NewHostEvent(host=ip, cloud=cloud))
+                self.publish_event(NewHostEvent(host=ip))
```

Comment:
Why did you remove the `cloud` parameter?

Reply:
this was the whole point of the issue, we don't need to check cloud type for each IP, we will only check it later when we access event.cloud